presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -889,14 +890,18 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
             }
         }
 
-        if (table.getPartitionColumns().isEmpty()) {
+        List<Column> partitionColumns = table.getPartitionColumns();
+
+        if (partitionColumns.isEmpty()) {
```

Comment:
The reason I avoided doing this is because some code in buildTableObject() - specifically lines 702-706 - are a necessary step prior to this check. Should I just duplicate those lines into beginCreateTable()?